src/main/java/nextstep/ladder2/domain/NxLadder.java

```diff
@@ -0,0 +1,35 @@
+package nextstep.ladder2.domain;
+
+import java.util.Collections;
+import java.util.List;
+import nextstep.ladder2.domain.builder.LadderBuilder;
+
+public class NxLadder {
+    private static final int LADDER_MIN_HEIGHT = 1;
+    private static final int LADDER_MIN_PEOPLE = 2;
+
+    private final List<NxLine> lines;
+
+    public NxLadder(int height, int countOfPerson, LadderBuilder ladderBuilder) {
+        if (height < LADDER_MIN_HEIGHT) {
+            throw new IllegalArgumentException("사다리 높이는 1 이상이어야 합니다: " + height);
+        }
+        if (countOfPerson < LADDER_MIN_PEOPLE) {
+            throw new IllegalArgumentException("참여자 수는 2 이상이어야 합니다: " + countOfPerson);
+        }
+        lines = ladderBuilder.build(height, countOfPerson);
+    }
+
+    public int ride(int startIndex) {
+        int updatedIndex = startIndex;
+        for (NxLine line : lines) {
+            NxBridge bridge = line.getBridgeByPoint(updatedIndex);
+            updatedIndex = bridge.movePosition(updatedIndex);
```

Comment:
방식을 변경해보면 어떨까요? 🤔

```suggestion
            updatedIndex = line.movePosition(updatedIndex);
```

이렇게 코드를 변경해도 동작 할 수 있게 해보면 좋을 것 같아요. :)

Reply:
관련 내용 반영하였습니다.